typing/datarepr.ml

```diff
@@ -48,6 +48,29 @@ let constructor_has_optional_shape ({cstr_attributes = attrs} : constructor_desc
   List.exists (fun (x,_) -> x.txt = internal_optional) attrs
 
 
+(**  [Types.constructor_description]
+     records the type at the definition type so for ['a option]
+     it will always be [Tvar]
+*)
+let cannot_inhabit_none_like_value (typ : Types.type_expr) =
```

Comment:
Should this be `cannot_be_of_option_type`.

Reply:
actually it is more strict than that. the value of this type can not inhabit value none, for example `Some Js.undefined` can not be compiled into `Js.undefined` directly